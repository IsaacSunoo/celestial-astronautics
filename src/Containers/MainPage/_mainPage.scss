.home-page {
    background-color: #000;
    color: #FFF;
    display: flex;
    flex-direction: column;
    position: relative;
    text-align: center;
    z-index: 1;

    h3 {
        padding-bottom: 30px;
    }

    .daily-date {
        padding-top: 30px;
        padding-bottom: 15px;
    }

    .center-media {
        margin-left: auto;
        margin-right: auto;
    }

    .daily-description {
        padding-left: 250px;
        padding-right: 250px;
    }

    img {
        max-width: 660px;
    }
}
