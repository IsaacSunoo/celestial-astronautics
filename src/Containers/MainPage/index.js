import React, { Component } from 'react';
import LoadingGif from '../../images/Loading.gif';
import { connect } from 'react-redux';
import ParticleContainer from '../../components/Particles';
import PropTypes from 'prop-types';

export class MainPage extends Component {
  render() {
    const { date, explanation, hdurl, title, media_type, url, isLoading } = this.props;
    return isLoading ? (
      <div className="loading-display">
        <h2>Loading</h2>
        <img src={LoadingGif} alt="loading gif" />
      </div>
    ) : (
      <div className="home-page">
        <ParticleContainer />
        <h3>{title}</h3>
        {media_type === 'video' ? (
          <iframe width="560" height="315" src={url} title={title} className="center-media" />
        ) : (
          <img src={hdurl} alt="daily nasa media" className="center-media" />
        )}
        <p className="daily-date">{date}</p>
        <p className="daily-description">{explanation}</p>
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  isLoading: state.isLoading
});

MainPage.propTypes = {
  isLoading: PropTypes.bool.isRequired
};


export default connect(mapStateToProps, null)(MainPage);