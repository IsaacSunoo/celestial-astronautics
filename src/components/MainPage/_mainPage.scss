.home-page {
    background-color: #000;
    color: #FFF;
    padding-top: 30px;
    padding-bottom: 300px;
}
