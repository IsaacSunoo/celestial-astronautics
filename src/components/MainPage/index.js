import React from 'react';

const MainPage = ({ date, explanation, hdurl, title, media_type, url }) => {
  return (
    <div className='home-page'>
      <h3>{title}</h3>
      <img src={hdurl} alt='daily nasa media' />
      {/* <iframe width='560' height='315' src={dailyNews.url}></iframe> */}
      <p>{date}</p>
      <p>{explanation}</p>
    </div>
  )
}

export default MainPage;