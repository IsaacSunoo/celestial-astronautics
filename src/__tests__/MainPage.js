import React from 'react';
import { MainPage, mapStateToProps } from '../containers/MainPage';
import { shallow } from 'enzyme';

const mockIsLoading = false;

describe('main page', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<MainPage isLoading={mockIsLoading} />);
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    describe('mapStateToProps', () => {
      it.skip('returns an object with isLoading', () => {
        const mockState = {
          isLoading: false,
          error: 'error'
        };
        const expectedState = {
          isLoading: false
        };
        const mappedProps = mapStateToProps(mockState);
        expect(mappedProps).toEqual(expectedState);
      });
    });
})