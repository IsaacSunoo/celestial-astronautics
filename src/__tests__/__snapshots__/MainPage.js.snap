// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main page should match snapshot 1`] = `
<div
  className="home-page"
>
  <ParticlesContainer />
  <h3 />
  <img
    alt="daily nasa media"
    className="center-media"
  />
  <p
    className="daily-date"
  />
  <p
    className="daily-description"
  />
</div>
`;
